src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiExtension.groovy

```diff
@@ -172,7 +174,7 @@ class JpiExtension {
                         [group: 'javax.servlet', name: servletApiArtifact, version: servletApiVersion],
                 )
 
-                compileOnly(jenkinsWar)
+                jenkinsWar = compileOnly(jenkinsWarCoordinates)
```

Comment:
Why is `jenkinsWar` needed for compilation? It is needed for running the tests and starting the server, maybe it should be part of `runtimeOnly`?

Reply:
Right. I think this is part of my misunderstanding of the `server` task. _jenkinsWar_ is already added to the test dependencies below. And it is directly used in the `server` task. I think this can just be removed here.

Reply:
Below however, it can probably be `testRuntimeOnly` (instead of `testImplementation`).